src/main/java/dev/blachut/svelte/lang/codeInsight/SvelteReferenceSearch.kt

```diff
@@ -0,0 +1,31 @@
+package dev.blachut.svelte.lang.codeInsight
+
+
+import com.intellij.lang.ecmascript6.psi.ES6ImportedBinding
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.psi.PsiReference
+import com.intellij.psi.search.*
+import com.intellij.psi.search.searches.ReferencesSearch
+import com.intellij.util.Processor
+import com.intellij.util.QueryExecutor
+
+class SvelteReferenceSearch : QueryExecutor<PsiReference, ReferencesSearch.SearchParameters> {
+    override fun execute(queryParameters: ReferencesSearch.SearchParameters, consumer: Processor<in PsiReference>): Boolean {
+        if (!ApplicationManager.getApplication().isReadAccessAllowed) {
+            return true
+        }
+        val elementToSearch = queryParameters.elementToSearch
```

Comment:
may be it is better to check that there is a svelte file? Because otherwise it will affect all js file inside the project.

Reply:
Absolutely, I'm new to plugin dev so thank you very much for the review!